File: matlab/mapc2m_torus.m
function [xp,yp,zp] = mapc2m_torus(xc,yc,alpha,beta)

if (nargin < 4)
    beta = 0;
end

pi2 = 2*pi;

r = alpha*(1 + beta*sin(pi2*xc));
R = 1 + r.*cos(pi2*yc);

xp = R.*cos(pi2*xc);
yp = R.*sin(pi2*xc);
zp = r.*sin(pi2*yc);

end

File: src/mappings/torus/mapc2m_torus.f
      subroutine mapc2m_torus(xc,yc,xp,yp,zp,alpha, beta)
      implicit none

      double precision xc,yc,xp,yp,zp
      double precision alpha, beta, r1, R


      double precision pi, pi2
      common /compi/ pi, pi2

      r1 = alpha*(1 + beta*sin(pi2*xc))
      R = 1 + r1*cos(pi2*yc)

      xp = R*cos(pi2*xc)
      yp = R*sin(pi2*xc)
      zp = r1*sin(pi2*yc)

      end

